lib/code_corps/policy/project_user.ex

```diff
@@ -2,16 +2,51 @@ defmodule CodeCorps.Policy.ProjectUser do
   @moduledoc """
   Handles `User` authorization of actions on `ProjectUser` records
   """
-  import CodeCorps.Policy.Helpers, only: [get_role: 1]
+
+  import CodeCorps.Policy.Helpers, only: [get_membership: 2, get_project: 1, get_role: 1]
 
   alias CodeCorps.{ProjectUser, Repo, User}
 
-  @spec create?(User.t, map) :: boolean
-  def create?(%User{id: user_id}, %{"user_id" => author_id})
-    when user_id == author_id and not is_nil(user_id), do: true
-  def create?(%User{}, %{}), do: false
+  @spec create?(User.t(), map) :: boolean
+  def create?(%User{id: user_id} = user, %{"user_id" => author_id, "project_id" => _} = params) do
+    user_membership = params |> get_project() |> get_membership(user)
+
+    user_role = user_membership |> get_role
+    new_role = Map.get(params, "role")
+
+    case [user_role, new_role] do
+      # Non-member can only make pending if they're the user
+      [nil, "pending"] ->
+        user_id == author_id
+
+      # No one else can do anything
+      ["pending", _] ->
+        false
+
+      ["contributor", _] ->
+        false
```

Comment:
A good approach if we want to make it extensible for other options down the line.

However, a simpler approach might be

```Elixir
def create?(%User{id: user_id}, %{"user_id" => member_id, "role" => "pending"}) when user_id == member_id, do: true
def create(%User{}, %{}), do: false
```

I think that works for all the cases we currently want to support, unless I'm missing something.